packages/cli/src/commonlib/codeFlowLogin.ts

```diff
@@ -298,6 +319,57 @@ export class CodeFlowLogin {
     }
   }
 
+  async getTokenByScopes(scopes: Array<string>, refresh = true): Promise<Result<string, FxError>> {
+    try {
+      if (!this.account) {
+        await this.reloadCache();
+      }
+      if (!this.account) {
+        const accessToken = await this.login();
+        return ok(accessToken);
+      } else {
+        return this.pca!.acquireTokenSilent({
+          account: this.account,
+          scopes: scopes,
+          forceRefresh: false,
+        })
+          .then((response) => {
+            if (response) {
+              return ok(response.accessToken);
+            } else {
+              return err(LoginCodeFlowError(new Error("No token response")));
+            }
+          })
+          .catch(async (error) => {
+            CliCodeLogInstance.necessaryLog(
+              LogLevel.Error,
+              "[Login] silent acquire token : " + error.message
+            );
+            if (!(await checkIsOnline())) {
+              return error(CheckOnlineError());
+            }
+            await this.logout();
+            (this.msalTokenCache as any).storage.setCache({});
```

Comment:
What is the behavior of `logout` and `setCache({})`? Will this clean up the token only for the specified scopes or all tokens in this cache?

Reply:
All scopes will be removed. Because multi `msal` versions may have different cache logic, they may work in one version but not work in the other version.